client/client.go

```diff
@@ -61,6 +63,16 @@ func NewClient(credentials *Credentials) (*Client, error) {
 		TLSClientConfig: &tls.Config{InsecureSkipVerify: credentials.Insecure}, // ignore expired SSL certificates
 	}
 
+	if credentials.ProxyURL != "" {
+		log.Printf("[DEBUG] Using proxy: %s\n", credentials.ProxyURL)
+		proxy, err := url.Parse(credentials.ProxyURL)
```

Comment:
WIth our previous discussions about this, it wouldn't be easier to add this logic to the mini-client and pull it from a `PROXY_URL` variable, it doesn't make a lot of sense to modify the provider, or at least to me it doesn't.